src/main/kotlin/com/milaboratory/mixcr/cli/ValidationUtil.kt

```diff
@@ -23,3 +25,27 @@ fun ValidationException.Companion.chainsExist(chains: Chains?, usedGenes: Collec
         }
     }
 }
+
+fun ValidationException.Companion.requireKnownSpecies(species: String?, vararg additional: String?) {
+    for (s in (listOf(species) + additional)) {
+        val known = try {
+            VDJCLibraryRegistry.getDefaultLibrary(s)
+            true
+        } catch (ignore: Throwable) {
+            false
+        }
+        require(known) { "Unknown species: $s" }
+    }
+}
+
+fun ValidationException.Companion.requireKnownChains(chains: String?, vararg additional: String?) {
+    for (s in (listOf(chains) + additional)) {
+        require(s != null && Chains.parse(s) != null) { "Unknown chains: $s" }
```

Comment:
Chains.parse is always notNull. Better to add validation to this point: com/milaboratory/mixcr/cli/Main.kt:341

Reply:
👍

Reply:
Removed validation at all, not needed for chains, they are arbitrary